### PR TITLE
Handle listing with marker after the last key

### PR DESCRIPTION
**Pull request by andrewgaul**
Also handle zero maxResults.  Found with Ceph s3-tests and s3proxy.

### CODE REVIEW

**Comment by andrewgaul** [comment]
@kahing Could you review this?

File: blobstore/src/main/java/org/jclouds/blobstore/config/LocalBlobStore.java [hunk 1]
```diff
@@ -262,7 +269,7 @@ public boolean apply(StorageMetadata o) {
          if (!contents.isEmpty()) {
             StorageMetadata lastElement = contents.last();
             contents = newTreeSet(Iterables.limit(contents, maxResults));
-            if (!contents.contains(lastElement)) {
+            if (maxResults != 0 && !contents.contains(lastElement)) {
```
Comment (kahing) [inline blobstore/src/main/java/org/jclouds/blobstore/config/LocalBlobStore.java]: can `maxResults` be negative?

**Comment by kahing** [comment]
+1

File: blobstore/src/main/java/org/jclouds/blobstore/config/LocalBlobStore.java (hunk 1 above)
Comment (andrewgaul) [inline blobstore/src/main/java/org/jclouds/blobstore/config/LocalBlobStore.java]: `ListOptions.maxResults` includes this check:

```java
checkArgument(maxKeys >= 0, "maxKeys must be >= 0");
```

**Pull request merged**